### PR TITLE
alacombe/expl-187/Logout HMSL when calling GGshield auth logout

### PR DESCRIPTION
**Pull request by iXioN**
Remove the JWT token when doing a ggshield logout

### CODE REVIEW

File: tests/unit/hmsl/test_hmsl_utils.py [hunk 1]
```diff
@@ -76,6 +81,20 @@ def test_get_token(isolated_fs, create_jwt_mock: MagicMock):
     assert token == "dummy_token"
 
 
+def test_remove_token(isolated_fs, create_jwt_mock: MagicMock):
+    """
+    GIVEN a logged in session to GIM
+    WHEN I want to remove the token
+    THEN the token as been removed
```
Comment (pierrelalanne) [inline tests/unit/hmsl/test_hmsl_utils.py:88]:
```suggestion
    THEN the token has been removed
```

File: tests/unit/hmsl/test_hmsl_utils.py [hunk 2]
```diff
@@ -76,6 +81,20 @@ def test_get_token(isolated_fs, create_jwt_mock: MagicMock):
     assert token == "dummy_token"
 
 
+def test_remove_token(isolated_fs, create_jwt_mock: MagicMock):
+    """
+    GIVEN a logged in session to GIM
+    WHEN I want to remove the token
```
Comment (pierrelalanne) [inline tests/unit/hmsl/test_hmsl_utils.py:87]: Nit ?
```suggestion
    WHEN I'm calling remove_token_from_disk
```

File: ggshield/hmsl/utils.py [hunk 3]
```diff
@@ -99,6 +102,14 @@ def load_token_from_disk() -> Optional[str]:
...
     try:
-        open(get_cache_dir() + "/hmsl_token", "w").write(token)
+        open(TOKEN_PATH, "w").write(token)
     except Exception as e:
         logger.warning(f"Error while saving token: {e}")
+
+
+def remove_token_from_disk() -> None:
+    try:
+        if os.path.exists(TOKEN_PATH):
```
Comment (pierrelalanne) [inline ggshield/hmsl/utils.py:112]: Nit: I would add an `except OSError` and pass instead of having an if.

File: ggshield/hmsl/utils.py [hunk 4]
```diff
@@ -18,6 +19,8 @@
 
 logger = logging.getLogger(__name__)
 
+TOKEN_PATH = "{}/hmsl_token".format(get_cache_dir())
```
Comment (pierrelalanne) [inline ggshield/hmsl/utils.py:22]: f-strings are more concise:  
```suggestion
TOKEN_PATH = f"{get_cache_dir()}/hmsl_token"
```

**Review by pierrelalanne — COMMENTED** [review]
A few remarks. Otherwise, looks good to me when testing.

**Review by Walz — CHANGES_REQUESTED** [review]
I've merged a fix for the tests, can you rebase and squash the branch. Also you are missing a changelog (`scriv create`).

**Review by Walz — APPROVED** [review]

**Review by Walz — APPROVED** [review]

**Pull request merged**